tests/phpunit/Console/E2ETest.php

```diff
@@ -120,14 +121,23 @@ protected function tearDown(): void
         chdir($this->cwd);
     }
 
+    /**
+     * Ensure this test belongs to e2e group. We exclude it during E2E testing
+     * from this class to avoid causing endless recursive testing loop.
+     */
+    public function test_it_belongs_to_e2e_group(): void
```

Comment:
I'm not convinced this one is necessary... You are basically testing that this class is marked with `@group e2e`. Unlike the `@group integration` which aimed to test that if detected an IO/network operation, it should be marked as `integration`

Reply:
Yes, that’s correct. It is marked as such. But also has to be marked e2e. Think this is wrong?

Reply:
I think if marked as `@group e2e` `@group integration` should be unnecessary: I admit did not think that one through when I added the integration checks.

I would remove the need for that test and also make more sense IMO

Reply:
Alright, I’ll try to remove this e2e group designation altogether then.

Reply:
It turns out @maks-rafalko needs `e2e` group because it is convenient. And it is. And I want this to be member `@group integration` because it is convenient and makes sense since this is a very IO intensive test. How do we do from here?